src/main/java/hudson/remoting/Channel.java

```diff
@@ -1004,7 +1030,7 @@ public void terminate(@Nonnull IOException e) {
             } // JENKINS-14909: leave synch block
         } finally {
             if (e instanceof OrderlyShutdown) e = null;
-            for (Listener l : listeners.toArray(new Listener[0])) {
+            for (Listener l : listeners) {
```

Comment:
woot. Could it be an excotic way of race condition protection?

Reply:
Not required anymore for sure

Reply:
Originally it was a `Vector`, so thread-safe to mutate but probably not for the iterator. `CopyOnWriteArrayList` is easier.